### PR TITLE
Convert hard-coded service account to dynamic one

### PR DESCRIPTION
**Pull request by heyselbi**
## Description

https://issues.redhat.com/browse/RHODS-7318
odh-model-controller is using hard-coded service account. PR is converting the service account name to a dynamic one. It checks if there is a service account name in custom ConfigMap. If not, it assigns a default service account name.

## How Has This Been Tested?

Not yet.

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

**Review by Jooho — COMMENTED** [review]
Test comment

File: controllers/inferecenceservice_serviceaccount.go [hunk 1]
```diff
@@ -160,7 +187,7 @@ func (r *OpenshiftInferenceServiceReconciler) reconcileSA(inferenceService *infe
 // ReconcileSA will manage the creation, update and deletion of the auth delegation SA + RBAC
 func (r *OpenshiftInferenceServiceReconciler) ReconcileSA(
 	inferenceservice *inferenceservicev1.InferenceService, ctx context.Context) error {
-	return r.reconcileSA(inferenceservice, ctx, newInferenceServiceSA)
+	return r.reconcileSA(inferenceservice, ctx)
```
[the remark this answers is not in the document]
Comment (heyselbi) [reply controllers/inferecenceservice_serviceaccount.go:183]: added the edits in the next commit

File: controllers/inferecenceservice_serviceaccount.go [hunk 2]
```diff
@@ -63,15 +54,51 @@ func createDelegateClusterRoleBinding(serviceAccountName string, serviceAccountN
...
+
+		// Extract the serviceAccountName key from the data
+		customServiceAccountName, exists := data["serviceAccountName"]
+		if !exists {
+			// Handle the case when the key is not present in the ConfigMap
+			log.Error("serviceAccountName key not found in 'model-serving-config' ConfigMap. Keeping default value.")
+		} else {
+			modelMeshServiceAccountName = customServiceAccountName
+		}
+	}
```
[the remark this answers is not in the document]
Comment (heyselbi) [reply controllers/inferecenceservice_serviceaccount.go]: added the edits in the next commit

File: controllers/inferecenceservice_serviceaccount.go [hunk 3]
```diff
@@ -63,15 +54,51 @@ func createDelegateClusterRoleBinding(serviceAccountName string, serviceAccountN
...
+		Namespace: inferenceService.Namespace,
+	}, customConfigMap)
+
+	modelMeshServiceAccountName := "modelmesh-serving-sa"
+
+	if err != nil {
+		if apierrs.IsNotFound(err) {
+			log.Info("Custom ConfigMap 'model-serving-config' not found. Using the default ServiceAccountName.")
+		} else {
+			log.Error(err, "Error trying to find 'model-serving-config' custom configmap.")
```
[the remark this answers is not in the document]
Comment (heyselbi) [reply controllers/inferecenceservice_serviceaccount.go]: added the edits in the next commit

**Comment by israel-hdez** [comment]
I barely remember, but I think the original ticket still stands.

**Comment by spolti** [comment]
Thanks @israel-hdez 
@heyselbi can you please update this PR?

**Comment by heyselbi** [comment]
I think @vaibhavjainwiz addressed it in a separate PR already. Closing it

**Pull request closed without merging**